android-json-form-wizard/src/test/java/com/vijay/jsonwizard/widgets/EditTextFactoryTest.java

```diff
@@ -0,0 +1,217 @@
+        List<View> viewList = factorySpy.getViewsFromJson("RandomStepName", context, formFragment, new JSONObject(gpsString), listener);
+        Assert.assertNotNull(viewList);
+        Assert.assertTrue(viewList.size() > 0);
+    }
+    @Test
+    public void testNameWithCumulativeTotalValidatorEditTextFactoryInstantiatesViewsCorrectly() throws Exception {
+        Assert.assertNotNull(factory);
+        EditTextFactory factorySpy = Mockito.spy(factory);
+        Assert.assertNotNull(factorySpy);
+
+        FormUtils formUtils = new FormUtils();
+        FormUtils formUtilsSpy = Mockito.spy(formUtils);
+        Assert.assertNotNull(formUtilsSpy);
+
+        Mockito.doReturn(resources).when(context).getResources();
+        Assert.assertNotNull(resources);
+
+        context.setTheme(R.style.NativeFormsAppTheme);
+        Mockito.doReturn(rootLayout).when(factorySpy).getRelativeLayout(context);
+        Assert.assertNotNull(rootLayout);
+
+        Mockito.doReturn(editTextLayout).when(rootLayout).findViewById(R.id.edit_text_layout);
+        Assert.assertNotNull(editTextLayout);
+
+        Mockito.doReturn(editText).when(editTextLayout).findViewById(R.id.edit_text);
+        Assert.assertNotNull(editText);
+
+        Mockito.doReturn(editButton).when(editTextLayout).findViewById(R.id.material_edit_text_edit_button);
+        Assert.assertNotNull(editButton);
+
+        Editable editable = new SpannableStringBuilder("Lifted");
+        Mockito.doReturn(editable).when(editText).getText();
+
+        Mockito.doReturn(jsonForm).when(formFragment).getCurrentJsonState();
+
+        Mockito.doReturn(relatedEditText1).when((JsonApi) context).getFormDataView("step1:medications_other");
+        Mockito.doReturn(relatedEditText2).when((JsonApi) context).getFormDataView("step1:medications_other2");
+        Mockito.doReturn(totalValueEditText).when((JsonApi) context).getFormDataView("step1:medications_other3");
+
+        String gpsString = "{\"key\":\"user_last_name\",\"openmrs_entity_parent\":\"\",\"openmrs_entity\":\"\",\"openmrs_entity_id\":\"\",\"type\":\"edit_text\",\"hint\":\"User Last name\",\"edit_type\":\"name\",\"value\":\"Lifted\",\"v_min_length\":{\"value\":\"6\",\"err\":\"Please enter a valid name\"},\"v_max_length\":{\"value\":\"10\",\"is_fixed_size\":\"true\",\"err\":\"Please enter a valid name\"},\"v_required\":{\"value\":\"true\",\"err\":\"Please enter the last name\"},\"v_cumulative_total\":{\"value\":\"medications_other3\",\"err\":\"Enter required fields\",\"related_fields\":[\"medications_other\",\"medications_other2\"]}}";
+        List<View> viewList = factorySpy.getViewsFromJson("step1", context, formFragment, new JSONObject(gpsString), listener);
+        Assert.assertNotNull(viewList);
+        Assert.assertTrue(viewList.size() > 0);
```

Comment:
I think we are better off asserting the exact value we expect based on the input in order to cover the ceiling edge case e.g. no duplicates